OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs

```diff
@@ -294,293 +278,283 @@ protected static Rectangle FlipRectangle(Rectangle rect, bool flipX, bool flipY)
-			tick = LoadField(Tick, data, defaults);
-			transpose = LoadField(Transpose, data, defaults);
-			frames = LoadField(Frames, data, defaults);
+
 			ignoreWorldTint = LoadField(IgnoreWorldTint, data, defaults);
 			scale = LoadField(Scale, data, defaults);
 
-			var flipX = LoadField(FlipX, data, defaults);
-			var flipY = LoadField(FlipY, data, defaults);
-			var zRamp = LoadField(ZRamp, data, defaults);
+			reverses = LoadField(Reverses, data, defaults);
+			transpose = LoadField(Transpose, data, defaults);
+			alpha = LoadField(Alpha, data, defaults);
+			alphaFade = LoadField(AlphaFade, data, defaults, out var alphaFadeLocation);
 
-			facings = LoadField(Facings, data, defaults);
-			interpolatedFacings = LoadField(InterpolatedFacings.Key, -1, data, defaults);
-			if (interpolatedFacings != -1 && (interpolatedFacings <= 1 || interpolatedFacings <= Math.Abs(facings) || interpolatedFacings > 1024
-				|| !Exts.IsPowerOf2(interpolatedFacings)))
-				throw new YamlException($"InterpolatedFacings must be greater than Facings, within the range of 2 to 1024, and a power of 2.");
+			var depthSprite = LoadField(DepthSprite, data, defaults, out var depthSpriteLocation);
+			if (!string.IsNullOrEmpty(depthSprite))
+				depthSpriteReservation = cache.ReserveSprites(depthSprite, new[] { LoadField(DepthSpriteFrame, data, defaults) }, depthSpriteLocation);
+
+			depthSpriteOffset = LoadField(DepthSpriteOffset, data, defaults);
 
 			if (facings < 0)
 			{
 				reverseFacings = true;
 				facings = -facings;
 			}
 
-			var offset = LoadField(Offset, data, defaults);
-			var blendMode = LoadField(BlendMode, data, defaults);
-
-			IEnumerable<int> GetUsedFrames(int frameCount)
-			{
-				if (LoadField(Length.Key, "", data, defaults) == "*")
-					length = frames?.Length ?? frameCount - start;
-				else
-					length = LoadField(Length, data, defaults);
+			if (interpolatedFacings != null && (interpolatedFacings < 2 || interpolatedFacings <= facings || interpolatedFacings > 1024 || !Exts.IsPowerOf2(interpolatedFacings.Value)))
+				throw new YamlException($"{interpolatedFacingsLocation}: {InterpolatedFacings.Key} must be greater than {Facings.Key}, within the range of 2 to 1024, and a power of 2.");
 
-				// Plays the animation forwards, and then in reverse
-				if (LoadField(Reverses, data, defaults))
-				{
-					var frames = this.frames != null ? this.frames.Skip(start).Take(length).ToArray() : Exts.MakeArray(length, i => start + i);
-					this.frames = frames.Concat(frames.Skip(1).Take(length - 2).Reverse()).ToArray();
-					length = 2 * length - 2;
-					start = 0;
-				}
+			if (length != null && length <= 0)
+				throw new YamlException($"{lengthLocation}: {Length.Key} must be positive.");
 
-				// Overrides Length with a custom stride
-				stride = LoadField(Stride.Key, length, data, defaults);
+			if (length == null && facings > 1)
+				throw new YamlException($"{facingsLocation}: {Facings.Key} cannot be used with {Length.Key}: *.");
```

Comment:
Should `length == null && facings > 1` use `length != null` given the message of `{Facings.Key} cannot be used with {Length.Key}`?

Reply:
`length = null` is used to represent "Length: *", i.e. load all the frames, so the current check is correct.